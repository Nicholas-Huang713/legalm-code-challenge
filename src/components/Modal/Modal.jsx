import "./Modal.scss";

const Modal = ({ onClose, children }) => {
  return (
    <div
      className="modal-backdrop"
      onClick={onClose}
      data-testid="modal-backdrop"
    >
      <div className="modal-content" onClick={(e) => e.stopPropagation()}>
        <button className="close-btn" onClick={onClose}>
          X
        </button>
        <section className="modal-content-main">{children}</section>
      </div>
    </div>
  );
};

export default Modal;
