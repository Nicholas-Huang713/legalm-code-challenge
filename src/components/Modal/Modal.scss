.modal-backdrop {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: rgba(0, 0, 0, 0.6);
    display: flex;
    justify-content: center;
    align-items: center;
    overflow-y: auto;
  }
  
  .modal-content {
    background-color: white;
    padding: 20px;
    border-radius: 8px;
    width: 90%;
    max-width: 600px;
    position: relative;
    color: black;
    overflow-y: auto;
  }

  .modal-content-main {
    height: 100%;
  }
  
  .close-btn {
    position: absolute;
    top: 10px;
    right: 10px;
    background: none;
    border: none;
    cursor: pointer;
    font-size: 16px;
    color: black;
  }
  
  @media (max-width: 768px) {
    .modal-content {
      width: 100%;
      max-width: 100%;
      border-radius: 0;
    }
  }
  