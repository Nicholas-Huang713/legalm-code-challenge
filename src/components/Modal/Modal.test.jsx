import { render, fireEvent, screen } from "@testing-library/react";
import Modal from "./Modal";

const renderModal = (isOpen, onClose) =>
  render(
    <Modal isOpen={isOpen} onClose={onClose}>
      <div>Children</div>
    </Modal>
  );
const mockOnClose = vi.fn();

describe("Modal component", () => {
  it("renders Modal", () => {
    renderModal(true, mockOnClose);
    expect(screen.getByText("Children")).toBeInTheDocument();
  });

  it("close btn calls mockOnClose", () => {
    renderModal(true, mockOnClose);
    const closeBtn = screen.getByText("X");
    fireEvent.click(closeBtn);
    expect(mockOnClose).toHaveBeenCalledTimes(1);
  });

  it("click backdrop calls mockOnClose", () => {
    renderModal(true, mockOnClose);
    const backdrop = screen.getByTestId("modal-backdrop");
    fireEvent.click(backdrop);
    expect(mockOnClose).toHaveBeenCalled();
  });
});
