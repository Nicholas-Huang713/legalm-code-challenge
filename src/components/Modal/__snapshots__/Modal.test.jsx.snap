// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal component > renders Modal 1`] = `
<div>
  <div
    class="modal-backdrop"
    data-testid="modal-backdrop"
  >
    <div
      class="modal-content"
    >
      <button
        class="close-btn"
      >
        X
      </button>
      <section
        class="modal-content-main"
      >
        <div>
          Children
        </div>
      </section>
    </div>
  </div>
</div>
`;
